addons/hr_calendar/models/res_partner.py

```diff
@@ -0,0 +1,112 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from pytz import UTC, timezone
+from datetime import datetime
+from collections import defaultdict
+from functools import reduce
+
+from odoo import api, models
+
+from odoo.osv import expression
+from odoo.addons.resource.models.utils import Intervals
+
+
+class Partner(models.Model):
+    _inherit = ['res.partner']
+
+    def _get_employees_from_attendees(self, everybody=False):
+        domain = [
+            ('company_id', 'in', self.env.companies.ids),
+            ('work_contact_id', '!=', False),
+        ]
+        if not everybody:
+            domain = expression.AND([
+                domain,
+                [('work_contact_id', 'in', self.ids)]
+            ])
+        return dict(self.env['hr.employee'].sudo()._read_group(domain, groupby=['work_contact_id'], aggregates=['id:recordset']))
+
+    def _get_schedule(self, start_period, stop_period, everybody=False, merge=True):
+        """
+        This method implements the general case where employees might have different resource calendars at different
+        times, even though this is not the case with only this module installed.
+        This way it will work with these other modules by just overriding
+        `_get_calendar_periods`.
+
+        :param start_period: the start datetime of the period
+        :param stop_period: the stop datetime of the period
+        :param everybody: boolean used to represent the "everybody" filter on calendar
+        :param merge: boolean used to specify if calendar's work_intervals needs to be merged
+        :return: schedule (merged or not) by partner
+        :rtype: defaultdict
```

Comment:
For parameters docstring, it's more readable and I guess more proper, if we don't mingle the type of the argument and the decription
```suggestion
        :param datetime start_period: the start of the period
        :param datetime stop_period: the stop of the period
        :param boolean everybody: represents the "everybody" filter on calendar
        :param boolean merge: specifies if calendar's work_intervals needs to be merged
        :return: schedule (merged or not) by partner
        :rtype: defaultdict
```